sources/config_reader.c

```diff
@@ -1062,6 +1062,15 @@ od_config_reader_import(od_config_t *config, od_rules_t *rules, od_error_t *erro
 		return -1;
 	rc = od_config_reader_parse(&reader);
 	od_config_reader_close(&reader);
+	if (rc == -1)
+		return -1;
+
+	//TODO: We should avoid doing here anything besides importing config.
```

Comment:
Please, address this TODO item.
This routine is called on SIGHUP, not only during initialization.